pkg/cache/v2/simple.go

```diff
@@ -81,6 +85,9 @@ type snapshotCache struct {
 	// hash is the hashing function for Envoy nodes
 	hash NodeHash
 
+	// heartbeatRoutines keeps track of the active heartbeat goroutines
+	heartbeatRoutines map[string]heartbeatHandle
```

Comment:
Why do we need so many go routines? Could we just have a single ticker that traverses the entire cache and re-pushes resources that aged?

Reply:
It just felt the easiest at the time, I'll give the single goroutine a go

Reply:
Is this field needed now? I don't see it being updated.